modules/snmp/validators.go

```diff
@@ -0,0 +1,144 @@
+package snmp
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+)
+
+func appendError(err error, msg string) error {
+	var e error
+	if err == nil {
+		if msg == "" {
+			return nil
+		}
+		e = errors.New(msg)
+	} else {
+		if msg == "" {
+			return err
+		}
+		e = fmt.Errorf("%w %s", err, msg)
+	}
+	return e
+}
+
+func (d Dimension) validateConfig(index_chart, index int) error {
+	var err error
+	err = nil
+	if d.Name == "" {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].name;", index_chart, index))
+	}
+	if d.OID == "" {
+		err = appendError(err, fmt.Sprintf("missing value: charts[%d].dimension[%d].oid;", index_chart, index))
+	}
+	if d.Algorithm == "" ||
+		(d.Algorithm != string(module.Incremental) &&
+			d.Algorithm != string(module.PercentOfIncremental) &&
+			d.Algorithm != string(module.PercentOfAbsolute)) {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].algorithm;", index_chart, index))
```

Comment:
An empty value is ok here (Netdata will use an internal default), it is better to have as many optional parameters as possible.

Reply:
The approach here is that, validator shall be strict. But any unset/default values must be handled at init() function.

Reply:
Mmm, what I mean is an empty (missing value) "Algorithm" is not an error/problem. It is valid.